File: app/src/main/java/com/test/tclick/MainActivity.java
package com.test.tclick;

import android.Manifest;
import android.accessibilityservice.AccessibilityService;
import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.provider.Settings;
import android.util.Log;
import android.view.View;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;
import androidx.core.content.ContextCompat;

public class MainActivity extends AppCompatActivity {

    private static final int REQUEST_PERMISSION = 1;
    private static final String TAG = "MainActivity";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Log.d(TAG, "onCreate: Activity created.");

        if (!isAccessibilityServiceEnabled(this, MyAccessibilityService.class)) {
            Log.d(TAG, "onCreate: Accessibility Service not enabled.");
            Toast.makeText(this, "Please enable the Accessibility Service.", Toast.LENGTH_LONG).show();
            startAccessibilitySettings();
        } else {
            Log.d(TAG, "onCreate: Accessibility Service already enabled.");
            checkAndRequestPermissions();
        }
    }

    private boolean isAccessibilityServiceEnabled(Context context, Class<? extends AccessibilityService> service) {
        String colonSplitter = ":";
        String colonSplitterValue = colonSplitter + service.getName() + colonSplitter;

        String enabledServices = Settings.Secure.getString(context.getContentResolver(),
                Settings.Secure.ENABLED_ACCESSIBILITY_SERVICES);
        boolean isEnabled = enabledServices.contains(colonSplitterValue);
        Log.d(TAG, "isAccessibilityServiceEnabled: " + isEnabled);
        return isEnabled;
    }

    private void startAccessibilitySettings() {
        Intent intent = new Intent(Settings.ACTION_ACCESSIBILITY_SETTINGS);
        startActivity(intent);
        Log.d(TAG, "startAccessibilitySettings: Accessibility settings opened.");
    }

    private void launchTossApp() {
        Intent launchIntent = getPackageManager().getLaunchIntentForPackage("viva.republica.toss");
        if (launchIntent != null) {
            startActivity(launchIntent);
            Log.d(TAG, "launchTossApp: Toss app launched.");
        } else {
            Log.e(TAG, "launchTossApp: Toss app not found.");
            Toast.makeText(this, "Toss app not found.", Toast.LENGTH_SHORT).show();
        }
    }

    private void checkAndRequestPermissions() {
        if (ContextCompat.checkSelfPermission(this, Manifest.permission.WRITE_EXTERNAL_STORAGE)
                != PackageManager.PERMISSION_GRANTED) {
            ActivityCompat.requestPermissions(this,
                    new String[]{Manifest.permission.WRITE_EXTERNAL_STORAGE},
                    REQUEST_PERMISSION);
            Log.d(TAG, "checkAndRequestPermissions: Permission requested.");
        } else {
            Log.d(TAG, "checkAndRequestPermissions: Permission already granted.");
            launchTossApp(); // 권한이 이미 허용된 경우 앱을 실행합니다.
        }
    }

    @Override
    public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
                                           @NonNull int[] grantResults) {
        super.onRequestPermissionsResult(requestCode, permissions, grantResults);
        if (requestCode == REQUEST_PERMISSION) {
            if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
                Log.d(TAG, "onRequestPermissionsResult: Permission granted.");
                Toast.makeText(this, "Permission granted!", Toast.LENGTH_SHORT).show();
                launchTossApp(); // 권한이 승인되었으므로 앱을 실행합니다.
            } else {
                Log.d(TAG, "onRequestPermissionsResult: Permission denied.");
                Toast.makeText(this, "Permission denied.", Toast.LENGTH_SHORT).show();
            }
        }
    }
    @Override
    protected void onResume() {
        super.onResume();
        if (isAccessibilityServiceEnabled(this, MyAccessibilityService.class)) {
            checkAndRequestPermissions();
        } else {
            Toast.makeText(this, "Please enable the Accessibility Service.", Toast.LENGTH_LONG).show();
            startAccessibilitySettings();
        }
    }
}

File: app/src/main/java/com/test/tclick/MainActivity.kt
package com.test.tclick

import android.accessibilityservice.AccessibilityService
import android.content.ComponentName
import android.content.Context
import android.content.DialogInterface
import android.content.Intent
import android.os.Bundle
import android.provider.Settings
import android.util.Log
import android.view.View
import android.widget.Button
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private var tossButton: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // 멤버 변수 tossButton에 버튼 인스턴스를 할당
        tossButton = findViewById(R.id.buttonToss)
        tossButton?.setOnClickListener(View.OnClickListener { v: View? -> triggerAccessibilityService() })
    }

    override fun onResume() {
        super.onResume()

        // 접근성 서비스가 활성화되어 있는지 확인
        if (!isAccessibilityServiceEnabled(this, MyAccessibilityService::class.java)) {
            // 접근성 서비스 활성화 안내 다이얼로그 표시
            AlertDialog.Builder(this)
                .setTitle("접근성 서비스 필요")
                .setMessage("이 기능을 사용하려면 접근성 서비스를 활성화해야 합니다.")
                .setPositiveButton("설정으로 이동") { dialog: DialogInterface?, which: Int ->
                    val intent = Intent(Settings.ACTION_ACCESSIBILITY_SETTINGS)
                    startActivity(intent)
                }
                .setNegativeButton("취소", null)
                .show()
            tossButton?.isEnabled = false
        } else {
            tossButton?.isEnabled = true
        }
    }

    // 접근성 서비스에 작업을 트리거하는 메서드
    private fun triggerAccessibilityService() {
        val intent = Intent(this, MyAccessibilityService::class.java)
        intent.setAction("com.test.tclick.ACTION_TRIGGER_ACCESSIBILITY")
        Log.d("MainActivity", "Starting service: " + intent.action)
        startForegroundService(intent)
    }

    // 접근성 서비스 활성화 여부 확인
    private fun isAccessibilityServiceEnabled(
        context: Context,
        service: Class<out AccessibilityService?>
    ): Boolean {
        val expectedComponentName = ComponentName(context, service)
        val enabledServices = Settings.Secure.getString(
            context.contentResolver, Settings.Secure.ENABLED_ACCESSIBILITY_SERVICES
        )
        if (enabledServices != null) {
            val componentNames =
                enabledServices.split(":".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
            for (componentName in componentNames) {
                val enabledService = ComponentName.unflattenFromString(componentName)
                if (enabledService != null && enabledService == expectedComponentName) {
                    return true
                }
            }
        }
        return false
    }
}
